src/expand.iced

```diff
@@ -0,0 +1,96 @@
+parse3 = require './parse3'
+pgp_utils = require('pgp-utils')
+{katch,json_stringify_sorted} = pgp_utils.util
+triplesec = require('triplesec')
+{ExpansionError} = require('./errors').errors
+{prng,createHmac} = require 'crypto'
+
+#========================================================
+
+check_expansions = ({expansions}) ->
+  for k,v of expansions
+    check_expansion_kv {k, v}
+
+#========================================================
+
+exports.hmac_obj = hmac_obj = ({obj, key}) ->
+  hmac = createHmac("sha256", key)
+  s = json_stringify_sorted obj
+  hmac.update(Buffer.from(s, 'utf8')).digest()
```

Comment:
this can have utf8?

Reply:
no i just meant ascii, is that possible with `Buffer.from`?

Reply:
i'll change to `ascii`